webapp/templates/proxy-create.xhtml

```diff
@@ -0,0 +1,303 @@
+            <input type="hidden" property="calli:alternate" content="?alternate" value="{?alternate}"/><!-- special case for viewing when no GET rule -->
+            <input type="hidden" property="calli:copy" content="?copy" value="{?copy}"/>
+            <input type="hidden" property="calli:post" content="?post" value="{?post}"/>
+            <input type="hidden" property="calli:put" content="?put" value="{?put}"/>
+            <input type="hidden" property="calli:patch" content="?patch" value="{?patch}"/>
+            <input type="hidden" property="calli:delete" content="?delete" value="{?delete}"/>            
+        </div>
+        <input id="cacheProperty" type="hidden" property="calli:cacheControl" onchange="calli.updateProperty(event, 'calli:cacheControl')"/>
+                
+        <div class="row">
+            <div id="rules" class="col-sm-12">
+                <label>Handlers</label>
+
+                <div class="panel panel-default rule" data-method="GET">
+                    <div class="panel-heading">
+                        <h3 class="panel-title">GET Handler
+                            <span class="pull-right">
+                                <a href="#getHandler" data-toggle="collapse" class="glyphicon glyphicon-minus"></a>
+                            </span>
+                        </h3>
+                    </div>
+                    <div class="panel-body collapse in" id="getHandler">
+                        <fieldset class="row">
+                            <div class="form-group col-sm-2">
+                                <label for="getActionType">Action</label>
+                                <div>
+                                    <select id="getActionType" class="form-control actionType">
+                                        <option value="calli:copy">Copy (200)</option>
+                                        <option value="calli:post">POST</option>
+                                        <option value="calli:put">PUT</option>
+                                        <option value="calli:patch">PATCH</option>
+                                        <option value="calli:delete">DELETE</option>
+                                    </select>
+                                </div>
+                            </div>
+                            <div class="form-group col-sm-7">
+                                <label for="getPattern">URL Pattern</label>
+                                <div>
+                                    <input type="text" class="form-control requestPattern" placeholder="Pattern for variables"/>
+                                </div>
+                            </div>
+                            <div class="form-group col-sm-3">
+                                <label for="cache">Cache control</label>
+                                <div>
+                                    <select id="cache" class="form-control">
```

Comment:
I fought a lot with this before ending up with my version but I tried again with this approach and got it working. The issue is that the option with 'property' defined (which is only used on the edit form) must also exist in the create form otherwise you get an error on saving because the triple is missing. When the create form loads, this option is absent since the property has no value but it seems to exist for the server side validation. I don't know if this is a new requirement on create templates but I couldn't see it documented. I think it would be worth adding a pattern for dealing with selects for values of a predicate - the current patterns deal with relationships which don't hit this issue.

On the edit form you now get an extra option for the existing value so as you say it needs cleaning up in javascript. It should be a duplicate of an existing option so I do this:
```
        var selectionValue = $('#cache>option:selected').val();
        if (selectionValue) {
            $('#cache>option:selected').remove();
            $('#cache').val(selectionValue);
        }
```